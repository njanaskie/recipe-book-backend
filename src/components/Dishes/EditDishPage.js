import React, { useReducer } from 'react'
import DishForm from './DishForm'
import IngredientsContext from '../../../context/ingredients-context'
import ingredientsReducer from '../../reducers/ingredients'
import DishesContext from '../../../context/dishes-context'
import dishesReducer from '../../reducers/dishes'
import FiltersContext from '../../../context/filters-context'
import filtersReducer from '../../reducers/filters'
import EditDishHome from './EditDishHome'

const EditDishPage = () => {
    const [ingredients, dispatch] = useReducer(ingredientsReducer, [])
    const [dishes, dishDispatch] = useReducer(dishesReducer, [])
    const [filters, filtersDispatch] = useReducer(filtersReducer, [])

    return (
        <FiltersContext.Provider value={{ filters, filtersDispatch }}>
            <DishesContext.Provider value={{ dishes, dishDispatch }} >
                <IngredientsContext.Provider value={{ ingredients, dispatch }}>
                    <h1>Edit Dish</h1>
                    <EditDishHome />
                </IngredientsContext.Provider>
            </DishesContext.Provider>
        </FiltersContext.Provider>
    )
}

export default EditDishPage