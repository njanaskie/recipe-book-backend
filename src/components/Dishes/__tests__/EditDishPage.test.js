import React from 'react'
import { shallow } from 'enzyme'
import EditDishPage from '../EditDishPage'

test('should render EditDishPage correctly',() => {
    const wrapper = shallow(<EditDishPage />)
    expect(wrapper).toMatchSnapshot()
})