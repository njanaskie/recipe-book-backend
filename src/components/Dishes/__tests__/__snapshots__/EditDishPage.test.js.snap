// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditDishPage correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dishDispatch": [Function],
      "dishes": Array [],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "ingredients": Array [],
      }
    }
  >
    <h1>
      Edit Dish
    </h1>
    <EditDishHome />
  </ContextProvider>
</ContextProvider>
`;
